docs/fsharp/style-guide/formatting.md

```diff
@@ -0,0 +1,506 @@
+        (convertVolumeToLiter x)
+        (convertVolumeUSPint x)
+        (convertVolumeImperialPint x)
+```
+
+Anonymous function arguments can be either on next line or with a dangling `fun` on the argument line:
+
+```fsharp
+// OK
+let printListWithOffset a list1 =
+    List.iter (fun elem ->
+        printfn "%d" (a + elem)) list1
+
+// Tolerable
+let printListWithOffset a list1 =
+    List.iter (
+        fun elem ->
+            printfn "%d" (a + elem)) list1
+```
+
+### Formatting infix operators
+
+Separate operators by spaces. Obvious exceptions to this rule are the `!` and `.` operators.
+
+Infix expressions are OK to lineup on same column:
+
+```fsharp
+acc +
+(Printf.sprintf "\t%s - %i\n\r"
+     x.IngredientName x.Quantity)
+
+let function1 arg1 arg2 arg3 arg4 =
+    arg1 + arg2 +
+    arg3 + arg4
+```
+
+### Formatting pipeline operators
+
+Pipeline `|>` should go at the start of a line immediately under the expression being operated on:
+
+```fsharp
+// OK
+let methods2 =
+    System.AppDomain.CurrentDomain.GetAssemblies()
+    |> List.ofArray
+    |> List.map (fun assm -> assm.GetTypes())
+    |> Array.concat
+    |> List.ofArray
+    |> List.map (fun t -> t.GetMethods())
+    |> Array.concat
+
+// OK
+let methods2 = System.AppDomain.CurrentDomain.GetAssemblies()
+               |> List.ofArray
+               |> List.map (fun assm -> assm.GetTypes())
+               |> Array.concat
+               |> List.ofArray
+               |> List.map (fun t -> t.GetMethods())
+               |> Array.concat
+
+// Not OK
+let methods2 = System.AppDomain.CurrentDomain.GetAssemblies()
+            |> List.ofArray
+            |> List.map (fun assm -> assm.GetTypes())
+            |> Array.concat
+            |> List.ofArray
+            |> List.map (fun t -> t.GetMethods())
+            |> Array.concat
+```
+
+### Formatting modules
+
+Code in a local module must be indented relative to the module, but code in a top-level module does not have to be indented. Namespace elements do not have to be indented.
+
+```fsharp
+// A is a top-level module.
+module A
+
+let function1 a b = a - b * b
+```
+
+```fsharp
+// A1 and A2 are local modules.
+module A1 =
+    let function1 a b = a*a + b*b
+
+module A2 =
+    let function2 a b = a*a - b*b
+```
+
+### Formatting object expressions and interfaces
+
+Object expressions and interfaces should be aligned in the same way with `member` being indented after 4 spaces.
+
+```fsharp
+let comparer =
+    { new IComparer<string> with
+          member x.Compare(s1, s2) =
+              let rev (s : String) =
+                  new String (Array.rev (s.ToCharArray()))
+              let reversed = rev s1 i
+              reversed.CompareTo (rev s2) }
+```
+
+### Formatting whitespace in expressions
+
+Avoid extraneous whitespace in F# expressions.
+
+```fsharp
+// OK
+spam (ham.[1])
+
+// Not OK
+spam ( ham.[ 1 ] )
+```
+
+Named arguments should also not have space surrounding the `=`:
```

Comment:
I find this to be contentious though I don't dislike it. Might be worth putting two args into the example in order to illustrate that commas as separators are delimiters and hence would have a space after them